subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -177,6 +197,46 @@ public boolean test(String element) {
         }
     }
 
+    private interface SymbolicLinkMappings {
+        String remapAbsolutePath(Path path);
+        @CheckReturnValue
+        SymbolicLinkMappings withNewMapping(String source, String target, RelativePathTracker currentRelativePathTracker);
+        Iterable<String> getRemappedSegments(Iterable<String> segments);
+    }
```

Comment:
I don't think there's a point in having a `private` interface for a single `private` inner class.

Reply:
There are two implementations of the interface: The empty remapper and the other ones.